File: go/wam/478.go
const WamE2eMessageRecv = 478 // channel: regular

type E2eMessageRecvEvent struct {
	E2eSuccessful bool `wam:"e2e_successful"` // field 1
	E2eFailureReason E2EFAILUREREASON `wam:"e2e_failure_reason"` // field 2
	RetryCount int64 `wam:"retry_count"` // field 3
	E2eDestination E2EDESTINATION `wam:"e2e_destination"` // field 4
	E2eCiphertextType E2ECIPHERTEXTTYPE `wam:"e2e_ciphertext_type"` // field 5
	E2eCiphertextVersion int64 `wam:"e2e_ciphertext_version"` // field 6
	MessageMediaType MEDIATYPE `wam:"message_media_type"` // field 7
	E2eSenderType E2EDEVICETYPE `wam:"e2e_sender_type"` // field 8
	Offline bool `wam:"offline"` // field 9
	RevokeType REVOKETYPE `wam:"revoke_type"` // field 10
	IsLid bool `wam:"is_lid"` // field 11
	TypeOfGroup TYPEOFGROUPENUM `wam:"type_of_group"` // field 12
	EditType EDITTYPE `wam:"edit_type"` // field 13
	StanzaType STANZATYPE `wam:"stanza_type"` // field 14
	AgentEngagementType AGENTENGAGEMENTENUMTYPE `wam:"agent_engagement_type"` // field 15
	LocalAddressingMode ADDRESSINGMODE `wam:"local_addressing_mode"` // field 16
	MessageAddressingMode ADDRESSINGMODE `wam:"message_addressing_mode"` // field 17
	ServerAddressingMode ADDRESSINGMODE `wam:"server_addressing_mode"` // field 18
	BotType BOTTYPE `wam:"bot_type"` // field 19
	IsHostedChat bool `wam:"is_hosted_chat"` // field 20
	MessageDistributionType MESSAGEDISTRIBUTIONENUMTYPE `wam:"message_distribution_type"` // field 21
	InvisibleMessageCategory INVISIBLEMESSAGECATEGORYTYPE `wam:"invisible_message_category"` // field 22
	GroupEncryptionState GROUPENCRYPTIONTYPE `wam:"group_encryption_state"` // field 23
	EncryptionType ENCRYPTIONTYPECODE `wam:"encryption_type"` // field 25
	IsSimpleSignal bool `wam:"is_simple_signal"` // field 26
	IsPq bool `wam:"is_pq"` // field 27
	SessionScope SESSIONSCOPETYPE `wam:"session_scope"` // field 28
	ProcessingDeferred bool `wam:"processing_deferred"` // field 29
	SenderPlatform PLATFORMTYPE `wam:"sender_platform"` // field 30
	TraceIdInt int64 `wam:"trace_id_int"` // field 31
}

File: rust/wam/478.rs
pub const WAM_E2E_MESSAGE_RECV: u32 = 478;

#[derive(Debug, Default)]
pub struct E2eMessageRecvEvent {
    /// field 1, wire `e2e_successful`
    pub e2e_successful: Option<bool>,
    /// field 2, wire `e2e_failure_reason`
    pub e2e_failure_reason: Option<E2EFAILUREREASON>,
    /// field 3, wire `retry_count`
    pub retry_count: Option<i64>,
    /// field 4, wire `e2e_destination`
    pub e2e_destination: Option<E2EDESTINATION>,
    /// field 5, wire `e2e_ciphertext_type`
    pub e2e_ciphertext_type: Option<E2ECIPHERTEXTTYPE>,
    /// field 6, wire `e2e_ciphertext_version`
    pub e2e_ciphertext_version: Option<i64>,
    /// field 7, wire `message_media_type`
    pub message_media_type: Option<MEDIATYPE>,
    /// field 8, wire `e2e_sender_type`
    pub e2e_sender_type: Option<E2EDEVICETYPE>,
    /// field 9, wire `offline`
    pub offline: Option<bool>,
    /// field 10, wire `revoke_type`
    pub revoke_type: Option<REVOKETYPE>,
    /// field 11, wire `is_lid`
    pub is_lid: Option<bool>,
    /// field 12, wire `type_of_group`
    pub type_of_group: Option<TYPEOFGROUPENUM>,
    /// field 13, wire `edit_type`
    pub edit_type: Option<EDITTYPE>,
    /// field 14, wire `stanza_type`
    pub stanza_type: Option<STANZATYPE>,
    /// field 15, wire `agent_engagement_type`
    pub agent_engagement_type: Option<AGENTENGAGEMENTENUMTYPE>,
    /// field 16, wire `local_addressing_mode`
    pub local_addressing_mode: Option<ADDRESSINGMODE>,
    /// field 17, wire `message_addressing_mode`
    pub message_addressing_mode: Option<ADDRESSINGMODE>,
    /// field 18, wire `server_addressing_mode`
    pub server_addressing_mode: Option<ADDRESSINGMODE>,
    /// field 19, wire `bot_type`
    pub bot_type: Option<BOTTYPE>,
    /// field 20, wire `is_hosted_chat`
    pub is_hosted_chat: Option<bool>,
    /// field 21, wire `message_distribution_type`
    pub message_distribution_type: Option<MESSAGEDISTRIBUTIONENUMTYPE>,
    /// field 22, wire `invisible_message_category`
    pub invisible_message_category: Option<INVISIBLEMESSAGECATEGORYTYPE>,
    /// field 23, wire `group_encryption_state`
    pub group_encryption_state: Option<GROUPENCRYPTIONTYPE>,
    /// field 25, wire `encryption_type`
    pub encryption_type: Option<ENCRYPTIONTYPECODE>,
    /// field 26, wire `is_simple_signal`
    pub is_simple_signal: Option<bool>,
    /// field 27, wire `is_pq`
    pub is_pq: Option<bool>,
    /// field 28, wire `session_scope`
    pub session_scope: Option<SESSIONSCOPETYPE>,
    /// field 29, wire `processing_deferred`
    pub processing_deferred: Option<bool>,
    /// field 30, wire `sender_platform`
    pub sender_platform: Option<PLATFORMTYPE>,
    /// field 31, wire `trace_id_int`
    pub trace_id_int: Option<i64>,
}